client/src/utils/sidebarFunctionality.jsx

```diff
@@ -0,0 +1,281 @@
+};
+
+/**
+ * @description Gets called when a new application was selected in the dropwdown in the sidebar.
+ * Updates the state of the component and gets the tasks of the application for the TaskDropdown and clears the TaskDropdown.
+ * @param {Object} value new value of the ApplicationDropdown
+ * @param {String} robotId id of the robot
+ * @param {Object} elementState state of the selected element
+ * @param {Object} setterObject object containing the functions for setting the state in the React component
+ */
+const applicationChangedHandler = (
+  value,
+  robotId,
+  elementState,
+  setterObject
+) => {
+  setterObject.setElementState({
+    selectedElements: elementState.selectedElements,
+    currentElement: elementState.currentElement,
+  });
+
+  setterObject.setSelectedApplication(value);
+  resetRpaApplication(robotId, elementState.currentElement.id, value);
+  getTasksForApplication(value, setterObject);
+
+  setterObject.setOutputVariableName(undefined);
+  setterObject.setvariableList([]);
+};
+
+/**
+ * @description Gets called when a new task was selected in the dropwdown in the sidebar. Updates the state of the component
+ * and gets the parameters of the task and updates the XML RPA properties (adds the application and the task).
+ * @param {Object} value new value of the TaskDropdown
+ * @param {String} activityId id of the activity selected
+ * @param {String} robotId id of the robot
+ * @param {String} selectedApplication application selected
+ * @param {Object} setterObject object containing the functions for setting the state in the React component
+ */
+const taskChangedHandler = (
+  value,
+  activityId,
+  robotId,
+  selectedApplication,
+  setterObject
+) => {
+  setRpaTask(robotId, activityId, selectedApplication, value);
+  if (value) updateParamSection(activityId, robotId, setterObject);
+};
+
+/**
+ * @description Gets called when the value in a single input field for the parameters has been changed and updates
+ * the values in the ssot
+ * @param {String} activityId id of the activity selected
+ * @param {Object} value new value of input field
+ */
+const handleInputParameterChange = (activityId, value) => {
+  setSingleParameter(activityId, value);
+};
+
+/**
+ * @description Gets called when the name of the output variable has been changed and updates
+ * the output variables name in the ssot
+ * @param {String} activityId id of the activity selected
+ * @param {Object} newValue new value of the output variables name
+ */
+const handleOutputVarNameChange = (activityId, newValue) => {
+  setOutputValueName(activityId, newValue);
+};
+
+/**
+ * @description Gets called when the the button is pressed to save to the cloud.
+ * This function will retrieve the xml from the parser, parse that xml to a ssot and write the
+ * resulting ssot into the sessionStorage.
+ * @param {Object} modeler the modeling object
+ * @param {String} robotId id of the robot
+ */
+const onSaveToCloud = async (modeler, robotId) => {
```

Comment:
This function is not exclusively used by the modeler sidebar and should therefore be put in an other utils file

Reply:
Download robot file would be another one of those methods potentially used also somewhere else. I pulled those two methods into a new file